perf/benchmark/runner/prom.py

```diff
@@ -64,24 +63,67 @@ def __init__(
             self.headers["Host"] = host
         self.aggregate = aggregate
 
-    def fetch(self, query, groupby=None, xform=None):
+    def fetch_by_query(self, query):
         resp = requests.get(self.url + "/api/v1/query_range", {
             "query": query,
             "start": self.start,
             "end": self.end,
-            "step": "15"
+            "step": 15
         }, headers=self.headers)
 
         if not resp.ok:
-            raise Exception(str(resp.text))
+            raise Exception(str(resp))
+
+        return resp.json()
 
-        data = resp.json()
+    # We query from start_time to end_time and make 15s as the time interval to note down a data point
+    # The function is to calculate the average of all the data points we get
+    def get_average_within_query_time_range(self, data, resource_type):
+        if data["data"]["result"]:
+            data_points_list = data["data"]["result"][0]["values"]
+            data_sum = 0
+            for data_point in data_points_list:
+                data_sum += float(data_point[1])
+            data_avg = float(data_sum / len(data_points_list))
+            print(data_avg)
+            if resource_type == "cpu":
+                return to_mili_cpus(data_avg)
+            else:
+                return to_mega_bytes(data_avg)
+        return -1
+
+    def fetch(self, query, groupby=None, xform=None):
+        data = self.fetch_by_query(query)
         return compute_min_max_avg(
             data,
             groupby=groupby,
             xform=xform,
             aggregate=self.aggregate)
 
+    def fetch_istio_proxy_cpu_usage(self):
+        cpu_query = 'sum(rate(container_cpu_usage_seconds_total{job="kubernetes-cadvisor",container_name="istio-proxy"}[1m])) by (container_name)'
+        data = self.fetch_by_query(cpu_query)
+        avg_cpu = self.get_average_within_query_time_range(data, "cpu")
+        if avg_cpu == -1:
+            return []
+        return avg_cpu
+
+    def fetch_istio_proxy_memory_usage(self):
+        mem_query = 'container_memory_usage_bytes{job = "kubernetes-cadvisor", container_name="istio-proxy"}'
+        data = self.fetch_by_query(mem_query)
+        print(data)
```

Comment:
do we need this?

Reply:
Oops, need to remove this debugging info. Will do it in next PR at the same time.